src/backend/joanie/core/serializers/admin.py

```diff
@@ -1269,6 +1271,316 @@ def get_owner_name(self, instance) -> str:
+        otherwise an empty string.
+        """
+        return self.get_installment_value(instance, 1, "amount")
+
+    def get_installment_state_1(self, instance) -> str:
+        """
+        Return the state of the first installment if available,
+        otherwise an empty string.
+        """
+        return self.get_installment_value(instance, 1, "state")
+
+    def get_installment_due_date_2(self, instance) -> str:
+        """
+        Return the due date of the second installment if available,
+        otherwise an empty string.
+        """
+        return self.get_installment_value(instance, 2, "due_date")
+
+    def get_installment_amount_2(self, instance) -> str:
+        """
+        Return the amount of the second installment if available,
+        otherwise an empty string.
+        """
+        return self.get_installment_value(instance, 2, "amount")
+
+    def get_installment_state_2(self, instance) -> str:
+        """
+        Return the state of the second installment if available,
+        otherwise an empty string.
+        """
+        return self.get_installment_value(instance, 2, "state")
+
+    def get_installment_due_date_3(self, instance) -> str:
+        """
+        Return the due date of the third installment if available,
+        otherwise an empty string.
+        """
+        return self.get_installment_value(instance, 3, "due_date")
+
+    def get_installment_amount_3(self, instance) -> str:
+        """
+        Return the amount of the third installment if available,
+        otherwise an empty string.
+        """
+        return self.get_installment_value(instance, 3, "amount")
+
+    def get_installment_state_3(self, instance) -> str:
+        """
+        Return the state of the third installment if available,
+        otherwise an empty string.
+        """
+        return self.get_installment_value(instance, 3, "state")
+
+    def get_installment_due_date_4(self, instance) -> str:
+        """
+        Return the due date of the fourth installment if available,
+        otherwise an empty string.
+        """
+        return self.get_installment_value(instance, 4, "due_date")
+
+    def get_installment_amount_4(self, instance) -> str:
+        """
+        Return the amount of the fourth installment if available,
+        otherwise an empty string.
+        """
+        return self.get_installment_value(instance, 4, "amount")
+
+    def get_installment_state_4(self, instance) -> str:
+        """
+        Return the state of the fourth installment if available,
+        otherwise an empty string.
+        """
+        return self.get_installment_value(instance, 4, "state")
+
+
+class AdminOrderListExportSerializer(serializers.ListSerializer):
+    """
+    Serializer for exporting a list of orders to a CSV stream.
+    """
+
+    def update(self, instance, validated_data):
+        """
+        Only there to avoid a NotImplementedError.
+        """
+
+    @property
+    def csv_stream(self):
```

Comment:
I find it odd to have a property that is a generator.
You have to iterate over it, and it can only be iterated over once,
I am wondering if it is the reason of having the lines not streamed one by one?

Reply:
I removed the `property` decorator, but unfortunately the lines are still not streamed.